retrieval/k8s/k8s_stats.go

```diff
@@ -124,135 +132,189 @@ func writeK8sResourceFile(workDir *os.File, resourceName string,
 	return err
 }
 
+//nolint:gocyclo
+func shouldSkipResource(k8Resource interface{}) bool {
```

Comment:
love this idea but prob want to run it by Peter first

Reply:
100% agreed, i've wanted to add something similar in our pipeline for some time now. mainly just want to make sure all parties are aware